otherlibs/stdune-unstable/temp.ml

```diff
@@ -2,39 +2,41 @@ type what =
   | Dir
   | File
 
-let try_times n ~f =
+let prng = lazy (Random.State.make_self_init ())
+
+let try_paths n ~dir ~prefix ~suffix ~f =
```

Comment:
I still don't like this helper very much. In retrospect, this should have been written as:

```
val random_paths : dir:Path.t -> prefix:string -> suffix:string -> Path.t Seq.t
```

And then callers could use it as:

```ocaml
random_paths ~dir ~prefix ~suffix |> Seq.take 1000 |> Seq.find ~f:Path.exists
```

etc.

This isn't pressing however.

Reply:
Yes, we can probably improve this in future.

Reply:
Btw, I'm going to add `with_temp_dir` in another PR, and I'll need this function again. I'll see if I can improve it.